### PR TITLE
fix: fix for failing rpm build

### PR DESCRIPTION
**Pull request by kridai**
# Fixes #

Fixed failing RPM build. 
- Previously we were using `centos:latest` tag in Dockerfile, This was picking up centos8. Centos has shifted focus towards  supporting centos stream. More on this [here](https://www.centos.org/cl-vs-cs/). Since centos is updating the centos stream image found [here](https://wiki.centos.org/FAQ/CentOSStream#What_artifacts_are_built.3F). Pulling it in docker file
- The sym linking for twilio wasn't working for this version, so updated twilio.spec file.
- Added the postinstall script which was missing

Testing
- Local : logs https://gist.github.com/kridai/7cfa368fcf0b782dda7ef1a4709aaecd
- action: https://github.com/kridai/twilio-cli/runs/5142627518?check_suite_focus=true
- generated package: https://github.com/kridai/twilio-cli/releases/tag/3.1.0

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [ ] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-cli/blob/main/CONTRIBUTING.md) and my PR follows them
- [ ] I have titled the PR appropriately
- [ ] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

File: .github/rpm/twilio.spec [hunk 1]
```diff
@@ -34,13 +34,19 @@ rm -rf %{buildroot}/%{name}-%{version}
...
 rm -rf %{buildroot}/%{name}-%{version}
 rm -rf $RPM_BUILD_DIR
 
+%post
+ln -sf /usr/local/lib/%{name}/bin/%{name} /usr/local/bin/%{name}
+cd /usr/local/lib/%{name}
+PATH=$PATH:$PWD/bin eval $(PATH=$PATH:$PWD/bin node -p "require('./package').scripts.postinstall")
+
+%postun
+rm -f /usr/local/bin/%{name}
```
Comment (kridai) [inline .github/rpm/twilio.spec:48]: removing the symlink as part of post uninstall process

File: .github/rpm/twilio.spec [hunk 2]
```diff
@@ -34,13 +34,19 @@ rm -rf %{buildroot}/%{name}-%{version}
...
 cp -a $RPM_BUILD_DIR/%{name}/* %{buildroot}/usr/local/lib/%{name}
-cd %{buildroot}
-ln -sf /usr/local/lib/twilio/bin/twilio /usr/local/bin/twilio
 
 %clean
 rm -rf %{buildroot}/%{name}-%{version}
 rm -rf $RPM_BUILD_DIR
 
+%post
+ln -sf /usr/local/lib/%{name}/bin/%{name} /usr/local/bin/%{name}
```
Comment (kridai) [inline .github/rpm/twilio.spec:43]: adding symlinking as part of postInstall script

File: .github/rpm/twilio.spec [hunk 3]
```diff
@@ -34,13 +34,19 @@ rm -rf %{buildroot}/%{name}-%{version}
...
+cd /usr/local/lib/%{name}
+PATH=$PATH:$PWD/bin eval $(PATH=$PATH:$PWD/bin node -p "require('./package').scripts.postinstall")
```
Comment (kridai) [inline .github/rpm/twilio.spec:45]: running postinstall script defined in package.json

File: .github/actions/build-rpm/Dockerfile [hunk 4]
```diff
@@ -6,7 +6,7 @@ RUN npm install && \
     npm run build && \
     npm run package
 
-FROM centos:latest
+FROM quay.io/centos/centos:stream8
```
Comment (kridai) [inline .github/actions/build-rpm/Dockerfile:9]: update to use centos stream 8, support is there till [2024-05-31](https://www.centos.org/cl-vs-cs/)

**Review by shrutiburman — APPROVED** [review]

**Review by shamantraghav — APPROVED** [review]

**Pull request merged**